File: WebApi/AngularUniversity/src/app/university-app/shared/faculty.service.ts
import { Injectable } from "@angular/core";
import { IFaculty } from './faculty.interface';
import { Observable } from "rxjs";
import { HttpClient } from "@angular/common/http";

@Injectable()
export class FacultyService {
    private readonly apiUrl: string = 'http://localhost:4200/rest/faculty';

    constructor(private httpClient: HttpClient) {
    }

    public addFaculty(faculty: IFaculty): Observable<null> {
        
        return this.httpClient.post<null>(`${this.apiUrl}/create`, faculty);
    }

    public deleteFaculty(id: number): Observable<object> {
        return this.httpClient.delete(`${this.apiUrl}/delete/${id}`);
    }

    public getFaculties(): Observable<IFaculty[]> {
        return this.httpClient.get<IFaculty[]>(`${this.apiUrl}/all`);
    }

    public updateFaculty(faculty: IFaculty): Observable<null> {
        return this.httpClient.post<null>(`${this.apiUrl}/update`, faculty);
    }

    public getFacultyByName(name: string): Observable<IFaculty> {
        return this.httpClient.get<IFaculty>(`${this.apiUrl}/name/${name}`);
    }
}

File: WebApi/WebAppUniversity/Services/FacultyService.cs
using WebAppUniversity.Domain;
using WebAppUniversity.Dto;
using WebAppUniversity.Repositories;
using WebAppUniversity.UnitOfWork;

namespace WebAppUniversity.Services
{
    public class FacultyService : IFacultyService
    {
        private readonly IFacultyRepository _facultyRepository;
        private readonly IUnitOfWork _unitOfWork;

        public FacultyService(IFacultyRepository facultyRepository, IUnitOfWork unitOfWork)
        {
            _facultyRepository = facultyRepository;
            _unitOfWork = unitOfWork;
        }

        public List<Faculty> GetFaculties()
        {
            return _facultyRepository.GetFaculties();
        }

        public int CreateFaculty(FacultyDto faculty)
        {
            if (faculty == null)
            {
                throw new Exception($"{nameof(faculty)} not found");
            }

            Faculty facultyEntity = faculty.ConvertToFaculty();

            int id = _facultyRepository.Create( facultyEntity ).Id;
            _unitOfWork.Commit();

            return id;
        }

        public void DeleteFaculty(int facultyId)
        {
            Faculty faculty = _facultyRepository.Get(facultyId);
            if (faculty == null)
            {
                throw new Exception($"{nameof(faculty)} not found, #Id - {facultyId}");
            }

            _facultyRepository.Delete(faculty);
            _unitOfWork.Commit();
        }

        public Faculty GetFaculty(int facultyId)
        {
            Faculty faculty = _facultyRepository.Get(facultyId);
            if (faculty == null)
            {
                throw new Exception($"{nameof(faculty)} not found, #Id - {facultyId}");
            }

            return faculty;
        }
        
        public int UpdateFaculty(FacultyDto facultyDto)
        {
            if (facultyDto == null)
            {
                throw new Exception($"{nameof(facultyDto)} not found");
            }
            
            Faculty facultyEntity = facultyDto.ConvertToFaculty();

            _facultyRepository.Update( facultyEntity );
            _unitOfWork.Commit();
            
            return facultyEntity.Id;
        }

        public Faculty GetFacultyByName(string name)
        {
            Faculty faculty = _facultyRepository.GetByName(name);
            if (faculty == null)
            {
                throw new Exception($"{nameof(faculty)} not found, #surname - {name}");
            }

            return faculty;
        }
    }
}
